src/microsim/schema/optical_config/config.py

```diff
@@ -150,14 +150,17 @@ def illumination(self) -> Spectrum | None:
 
         This represents the spectrum of light source and all of the filters in the
         excitation path. If there are multiple light sources, they are combined.
+        Different light sources can have different powers. In that case light sources
+        are scaled by the respective light powers.
         """
         exc = self.excitation
         if self.lights:
             l0, *rest = self.lights
-            illum_spect = l0.spectrum
+            p0, *p_rest = (l.power or 1 for l in self.lights)
+            illum_spect = l0.spectrum * p0
             if rest:
-                for light in rest:
-                    illum_spect = illum_spect + light.spectrum
+                for light, power in zip(rest, p_rest, strict=False):
+                    illum_spect = illum_spect + light.spectrum * power
```

Comment:
let's add a `scaled_spectrum` method to the light source, then reduce this whole thing to:

```python
illum_spect = sum(light.scaled_spectrum() for light in self.lights)
```

```python
    def scaled_spectrum(self) -> Spectrum:
        """Return light source spectrum scaled to power, if present."""
        spectrum = self.spectrum
        if self.power is not None:
            spectrum = spectrum * self.power
        return spectrum
```